src/debugpy/adapter/clients.py

```diff
@@ -299,7 +296,11 @@ def property_or_debug_option(prop_name, flag_name):
                 '"program", "module", and "code" are mutually exclusive'
             )
 
-        args += request("args", json.array(unicode))
+        # Propagate "args" via CLI if and only if shell expansion is requested.
+        args_expansion = request("argsExpansion", json.enum("shell", "none", optional=True))
```

Comment:
Do we need to expose this in the extension? to provide completion or do we want to keep it internal.

Reply:
I think it's best to expose. Some users might need the old behavior, and without this flag, it's just not possible.